src/newtondynamics_physics/SysNewton.cpp

```diff
@@ -0,0 +1,428 @@
+
+            if (rReg.all_of<ACompSolidCollider>(nextChild))
+            {
+                auto* childNwtCollider = rReg.try_get<ACompNwtCollider>(nextChild);
+
+                NewtonCollision const *collision;
+
+                if (nullptr == childNwtCollider)
+                {
+                    // No Newton collider exists yet
+                    collision = NewtonCreateSphere(nwtWorld, 0.5f, 0, NULL);
+                    rReg.emplace<ACompNwtCollider>(nextChild, collision);
+                    ACompNwtCollider f(collision);
+                }
+                else
+                {
+                    // Already has Newton collider
+                    collision = childNwtCollider->collision();
+                }
+
+                // Set transform relative to root body
+                Matrix4 f = Matrix4::translation(childMatrix.translation());
+                NewtonCollisionSetMatrix(collision, f.data());
+
+                // Add body to compound collision
+                NewtonCompoundCollisionAddSubCollision(rCompound, collision);
+            }
+
+            find_colliders_recurse(rScene, childHeir.m_childFirst, childMatrix,
+                                   nwtWorld, rCompound);
+        }
+
+        nextChild = childHeir.m_siblingNext;
+    }
+}
+
+void SysNewton::create_body(ActiveScene& rScene, ActiveEnt ent,
+                            NewtonWorld const* nwtWorld)
+{
+    ActiveReg_t &rReg = rScene.get_registry();
+
+    auto const& entHier = rReg.get<ACompHierarchy>(ent);
+    auto const& entTransform = rReg.get<ACompTransform>(ent);
+    auto const* entShape = rReg.try_get<ACompShape>(ent);
+
+    if ((! rReg.all_of<ACompSolidCollider>(ent)) || (entShape == nullptr))
+    {
+        // Entity must have a collision shape to define the collision volume
+        return;
+    }
+
+    NewtonBody const* body;
+
+    switch (entShape->m_shape)
+    {
+    case osp::phys::ECollisionShape::COMBINED:
+    {
+        // Combine collision shapes from descendants
+
+        NewtonCollision* rCompound
+                = NewtonCreateCompoundCollision(nwtWorld, 0);
+
+        NewtonCompoundCollisionBeginAddRemove(rCompound);
+        find_colliders_recurse(rScene, entHier.m_childFirst, Matrix4{},
+                               nwtWorld, rCompound);
+        NewtonCompoundCollisionEndAddRemove(rCompound);
+
+        if (auto* entNwtBody = rReg.try_get<ACompNwtBody>(ent);
+            nullptr != entNwtBody)
+        {
+            body = entNwtBody->body();
+            NewtonBodySetCollision(body, rCompound);
+        }
+        else
+        {
+            Matrix4 identity{};
+
+            body = NewtonCreateDynamicBody(
+                        nwtWorld, rCompound, identity.data());
+            rReg.emplace<ACompNwtBody>(ent, body);
+        }
+
+        // this doesn't actually destroy, it decrements a reference count
+        NewtonDestroyCollision(rCompound);
```

Comment:
Yikes...
Can't this be done using a smart pointer?

Reply:
Newton does seem to have support to add custom allocate and free functions, but that's all I can find.

Reply:
My question was more along the line of making rCompound an std::unique_ptr with a custom deleter, or something of that nature.